core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -361,12 +361,14 @@ static struct tee_mmap_region *init_xlation_table(struct tee_mmap_region *mm,
 	do  {
 		uint64_t desc = UNSET_DESC;
 
-		if ((mm->va + mm->size <= base_va) ||
-		    core_mmu_is_dynamic_vaspace(mm)) {
-			/*
-			 * Area now after the region or area should not be
-			 * mapped at init so skip it
-			 */
+		/* Skip unmaped entries */
+		while (core_mmu_is_dynamic_vaspace(mm))
```

Comment:
On table termination (`mm->size` is null), `core_mmu_is_dynamic_vaspace(mm)` currently returns `false`, hence exiting the loop once reached as expected.But I feel it is not really reliable (i.e if one day one replaces `core_mmu_is_dynamic_vaspace(mm)` with `!core_mmu_is_static_vaspace(mm)`). 

This is why I would rather add an explicit test on `mm->size` in the while: `while (mm->size && core_mmu_is_dynamic_vaspace(mm)) mm++;`. 

But maybe that's too much anticipating  later changes. I leave you to judge on this. It would not be nice to have a buggy mapping setup because of such hidden weakness.